controllers/storagecluster/cephfilesystem.go

```diff
@@ -41,7 +42,17 @@ func (r *StorageClusterReconciler) newCephFilesystemInstances(initData *ocsv1.St
 		},
 	}
 
-	if initData.Spec.StorageProfiles == nil {
+	// Load StorageProfile list
+	// Get StorageProfile list in the StorageCluster's ns
+	spList := &ocsv1.StorageProfileList{}
+	spErr := r.Client.List(r.ctx, spList, &client.ListOptions{Namespace: initData.GetNamespace()})
+	if spErr != nil {
+		r.Log.Error(spErr, "Unable to list StorageProfile objects")
+	}
+	// Save StorageProfile list to be reconciled
+	r.StorageProfiles = *spList
```

Comment:
You don't need an intermediate list( and a copy operation), just list into the target variable.
```suggestion
	spErr := r.Client.List(r.ctx, &r.StorageProfiles, &client.ListOptions{Namespace: initData.GetNamespace()})
	if spErr != nil {
		r.Log.Error(spErr, "Unable to list StorageProfile objects")
	}
```